pkg/relabel/relabel_test.go

```diff
@@ -84,3 +88,30 @@ func TestRelabel_Run(t *testing.T) {
 		})
 	}
 }
+
+func TestRlabel_NewFromViper(t *testing.T) {
+	t.Run("returns error when yaml config contains unknown keys", func(t *testing.T) {
+		config := []byte(`
+eventRelabelConfigs:
+  eventRelabelConfigs:
+  - source_labels: ["url"]
+    regexP: ".*operationName=(.*)(&.*)?$"
+    target_label: operation_name
+    replacement: "$1" `)
+		viper.SetConfigType("yaml")
+		err := viper.ReadConfig(bytes.NewBuffer(config))
+		if err != nil {
```

Comment:
We already use the `assert` package so you can use just https://pkg.go.dev/github.com/stretchr/testify@v1.7.0/assert#NotNilf maybe?